sqlitecluster/SQLiteClusterMessenger.cpp

```diff
@@ -100,19 +101,22 @@ bool SQLiteClusterMessenger::runOnLeader(BedrockCommand& command) {
 
         // SParseURI expects a typical http or https scheme.
         string url = "http://" + leaderAddress;
-        string host, path;
         if (!SParseURI(url, host, path) || !SHostIsValid(host)) {
             return false;
         }
 
         // Start our escalation timing.
         command.escalationTimeUS = STimeNow();
 
-        try {
-            // TODO: Future improvement - socket pool so these are reused.
-            // TODO: Also, allow S_socket to take a parsed address instead of redoing all the parsing above.
-            s = unique_ptr<SHTTPSManager::Socket>(new SHTTPSManager::Socket(host, nullptr));
-        } catch (const SException& exception) {
+        // Get a socket from the pool.
+        {
+            lock_guard<mutex> lock(_socketPoolMutex);
+            if (!_socketPool || _socketPool->host != host) {
```

Comment:
What will happen if `_socketPool` does exist but `_socketPool->host != host` and we re-assign it here?

Reply:
The old one will get destroyed when the pointer releases it. See this url:
https://en.cppreference.com/w/cpp/memory/unique_ptr

(search the page for `The object is disposed of`)